spec/views/user/user.html.erb_spec.rb

```diff
@@ -0,0 +1,97 @@
+require 'rails_helper'
+require 'capybara/rails'
+require 'capybara/rspec'
+
+RSpec.describe 'User', type: :view do
+  include Capybara::DSL
+  before do
+    @user = User.create(Name: 'Jot', Photo: 'john.png', Bio: 'from Hungary', posts_counter: 1)
+    @user1 = User.create(Name: 'Johnt', Photo: 'john.png', Bio: 'from Hungary', posts_counter: 1)
+    @user2 = User.create(Name: 'Dee', Photo: 'dee.png', Bio: 'uganda', posts_counter: 2)
+    @post1 = @user1.posts.create(text: 'Post 1', comments_counter: 0, likes_counter: 0)
+    @post = Post.create(author_id: @user.id, title: 'Test Post', text: 'text body', comments_counter: 1,
+                        likes_counter: 1)
+  end
+
+  describe 'index page' do
+    it 'Display the user name for all users' do
+      visit users_path
+      expect(page).to have_content(@user1.Name)
+      page.has_content?(@user2.Name)
+    end
+
+    it 'Display profile pic of each user' do
+      visit users_path
+      page.has_content?(@user1.Photo)
+      expect(page).to have_css("img[src*='john.png']")
+    end
+
+    it 'Display the number of posts a user has written' do
+      visit users_path
+      page.has_content?(@user1.posts_counter)
+      expect(page).to have_content(@user2.posts_counter)
+    end
+
+    it 'Redirects to the user show page when a user is clikced' do
+      visit users_path
+      click_link @user1.Name
+      expect(page).to have_current_path(user_path(@user1))
+    end
+  end
+
+  describe 'user show page' do
+    before do
+      visit user_path(@user1.id)
+      visit user_path(@user2.id)
+    end
+
+    it "Should see the user's profile picture" do
+      page.has_content?(@user1.Photo)
+      page.has_css?('.img-fluid')
+      expect(page).to have_css("img[src*='dee']")
+    end
+
+    it "I can see the user's username" do
+      page.has_content?(@user1.Name)
+      expect(page).to have_content(@user2.Name)
+    end
+
+    it 'I can see the number of posts the user has written.' do
+      page.has_content?("Number of posts: #{@user1.posts_counter}")
+      expect(page).to have_content(@user2.posts_counter)
+    end
+
+    it "I can see the user's bio." do
+      page.has_css?('user_bio')
+      if @user2.Bio.present?
+        expect(page).to have_content(@user2.Bio)
+      else
+        page.has_content?('No Bio entered')
+      end
+    end
+
+    it "Should see the user's profile picture" do
+      page.has_content?(@user1.Photo)
+      page.has_css?('.img-fluid')
+      expect(page).to have_css("img[src*='dee']")
+    end
+
+    it "I can see the user's first 3 posts." do
+      @user1.recent_posts.limit(3).each do |post|
+        expect(page).to have_content(post.title)
+      end
+    end
+
```

Comment:
![image](https://github.com/FrankMugagga/blog_app/assets/93039359/7bd22e1b-eaa2-43ff-80d8-89f4ed1825a1)

- [ ] Great work on testing 👍 but i guess you missed this test case in which you only need to assert that a Show all posts button exists 🤔 Kindly implement this test case to follow the project requirements 😊